Source/Controllers/SubviewControllers/SPTablesList.m

```diff
@@ -1937,6 +2058,83 @@ - (void) clearFilter
 	[listFilterField setStringValue:@""];
 }
 
+/**
+ * Reset the current pinned tables
+ */
+- (void)initPinnedTables {
+
+    [pinnedTables removeAllObjects];
+    NSString * hostName = [mySQLConnection host];
+    NSString * databaseName = [mySQLConnection database];
+    NSArray *tablesToPin = [_SQLitePinnedTableManager getPinnedTablesWithHostName:hostName databaseName:databaseName];
+    if (tablesToPin.count > 0) {
+        [tables insertObject:@"PINNED" atIndex:0];
```

Comment:
What happens here if the _previous_ tablesToPin count was greater than zero and the PINNED object was already inserted? Maybe instead do a if tablesToPin.count > 0 and safeObjectAtIndex(0) (maybe function name wrong, something safeObject... away from Xcode right now) != "PINNED" then insert?
In other words, check if index 0 is already PINNED before inserting.
Also, does that phrase "PINNED" need to be localized in the few places its handled? Is table/view/procedure localized?

Reply:
I guess in general here, it may be worth making sure there aren't pinned tables already and handling cleanup if they are? Or is this function guaranteed to only be called once ever?